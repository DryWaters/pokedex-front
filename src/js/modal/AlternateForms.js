import React from 'react';
import PropTypes from 'prop-types';

import Form from './Form';

import '../../styles/modal/AlternateForms.css';

const propTypes = {
  selectedForm: PropTypes.number.isRequired,
  forms: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      image_path: PropTypes.string.isRequired,
    }).isRequired
  ).isRequired,
};

class AlternateForms extends React.Component {
  constructor (props) {
    super(props);

    this.state = {
      selectedForm: 0,
    };
  }

  render () {
    const allForms = this.props.forms.map((form, index) => {
      return <Form key={form.name}
                   selected={this.props.selectedForm === index}
                   onClick={() => {}}
                   {...this.props.forms[index]}
             />
    });

    return (
      <div className="form-container">
        {allForms}
      </div>
    );
  }

}

AlternateForms.propTypes = propTypes;

export default AlternateForms;
