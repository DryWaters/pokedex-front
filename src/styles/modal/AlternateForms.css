.form-container {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  width: 100px;
  justify-content: space-between;
  margin-left: auto;
  margin-right: auto;
}
