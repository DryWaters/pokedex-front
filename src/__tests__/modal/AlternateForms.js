import React from 'react';
import { shallow } from 'enzyme';

import AlternateForms from '../../js/modal/AlternateForms';

import ExampleJSON from '../../../local_data/example_detail.json';

describe('AlternateForms Component', () => {
  let props;
  beforeEach(() => {
    props = {
      selectedForm: 0,
      selectForm: (() => {}),
      forms: [{
        name: ExampleJSON.forms[0].name,
        image_path: ExampleJSON.forms[0].image_path,
      }],
    };
  });

  it('should render AlternateForms component without crashing', () => {
    shallow(<AlternateForms {...props} />);
  });

  it('should match AlternateForms component snapshot', () => {
    const wrapper = shallow(<AlternateForms {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
