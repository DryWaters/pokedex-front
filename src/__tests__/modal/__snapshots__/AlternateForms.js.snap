// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlternateForms Component should match AlternateForms component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AlternateForms
    forms={
        Array [
            Object {
              "image_path": "/sprites/pokemon/large/12.png",
              "name": "Butterfree",
            },
          ]
    }
    selectForm={[Function]}
    selectedForm={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Form
          image_path="/sprites/pokemon/large/12.png"
          name="Butterfree"
          onClick={[Function]}
          selected={true}
/>,
      ],
      "className": "form-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Butterfree",
        "nodeType": "class",
        "props": Object {
          "image_path": "/sprites/pokemon/large/12.png",
          "name": "Butterfree",
          "onClick": [Function],
          "selected": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Form
            image_path="/sprites/pokemon/large/12.png"
            name="Butterfree"
            onClick={[Function]}
            selected={true}
/>,
        ],
        "className": "form-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Butterfree",
          "nodeType": "class",
          "props": Object {
            "image_path": "/sprites/pokemon/large/12.png",
            "name": "Butterfree",
            "onClick": [Function],
            "selected": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
